Clarify active step in workflow by changing the icons. Fix #8484

sdmoran opened the pull request:
<!--
Thanks for contributing to Wagtail! 🎉

Before submitting, please review the [contributor guidelines](https://docs.wagtail.org/en/latest/contributing/index.html).
-->

Hi all, first PR for Wagtail (and open source at all)! Please let me know if I am missing anything.

Here is a pass at addressing issue #8484.
- For completed steps, solid instead of dashed lines in the timeline
- For the in progress step, changed the icon to a spinner (created a new spinner-slow icon)
- For the in progress step, added the `aria-current="step"` tag

More than happy to discuss other ideas for clarifying the active step :)
Thanks!

Screenshot of workflow status modal:
![image](https://user-images.githubusercontent.com/33131675/195229015-352b4499-9916-4138-a835-21b0cdd9f312.png)

_Please check the following:_

-   [X] Do the tests still pass?[^1]
-   [X] Does the code comply with the style guide?
    -   [X] Run `make lint` from the Wagtail root.
-   N/A [ ] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [ ] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [X] **Please list the exact browser and operating system versions you tested**:
        - Desktop: Firefox, chrome
        - Mobile: Safari
    -   [ ] **Please list which assistive technologies [^3] you tested**:
-   [ ] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

thibaudcolas commented on wagtail/admin/templates/wagtailadmin/workflows/workflow_status.html:
```diff
@@ -13,8 +13,8 @@
             {% with status=task.task_state.status status_display=task.task_state.get_status_display|default:not_started_text %}
-                <li class="workflow-timeline__item workflow-timeline__item--{{ status|default:"pending" }}">
-                    <div class="workflow-timeline__line"></div>
+                <li class="workflow-timeline__item workflow-timeline__item--{{ status|default:"pending" }}" {% if status == "in_progress" %}aria-current="step"{% endif %}>
```
Nice!

thibaudcolas reviewed (changes requested):
Hi @sdmoran! This is looking very promising :)

I like the switch to a different icon to convey this, but I don’t think we should introduce a new icon to our set just for this use case. And I think spinners should be reserved for actual "loading" of something on the page. There really should be another existing icon which we can reuse.

For example the following ones seem interesting to me:

- `clipboard-list`
- `repeat`
- Or we could even try repurposing one like `radio-full` (just a circle)

For the different line, I think it’s an interesting concept but too subtle to convey the difference we want to make apparent here. I also think the icon approach is good enough to stand alone.

gasman commented:
I think it's important to change the "future" steps to use a different icon from the tick too - perhaps the empty radio button, if we're using radio-full for the current state.

sdmoran commented:
Thanks for the feedback! Addressed your comments and updated the screenshot.
- Removed solid line styling
- Changed active step icon to `radio-full`
- Changed future step icon to `radio-empty`

thibaudcolas reviewed (approved):
Looking spot on @sdmoran! Thank you for taking the time to include a screenshot, this helps a lot.
